include/tvm/ir_operator.h

```diff
@@ -519,6 +529,26 @@ inline bool is_const_int(const Expr& x, int64_t value) {
   return false;
 }
 
+template <typename ValueType>
+inline bool is_const_value(const Expr& e, ValueType value) {
+  static_assert(std::is_integral<ValueType>::value,
+                "Comparison to non-integer values is forbidden.");
+  // This implementation was copy-pasted from HalideIR
+  if (const ir::IntImm* i = e.as<ir::IntImm>()) {
+    return i->value == value;
+  } else if (const ir::UIntImm* i = e.as<ir::UIntImm>()) {
+    return (value >= 0) && (i->value == (uint64_t)value);
+  } else if (const ir::FloatImm* i = e.as<ir::FloatImm>()) {
```

Comment:
Yes, here we allow comparing expressions of any type, but only to ints.  I made this function a template with a static assertion to forbid things like `is_const_value(e, 3.14)` (if the type of value were int64_t instead of a template parameter, it still would be possible to do this, with 3.14 being converted to 3. Modern compilers may issue a warning in this case though.).